mbed_lstools/platform_database.py

```diff
@@ -480,15 +489,101 @@ def remove(self, id, permanent=False, device_type='daplink', verbose_data=False)
 
                 return _modify_data_format(removed, verbose_data)
 
-    def update_from_web(self, url='https://os.mbed.com/api/v3/platforms/?format=json'):
+    def update_from_web(self, url=DEFAULT_UPDATE_URL):
+        """Update entries in the platform database from the API on the Mbed website.
+
+        If the update fails to contact the provided url, a `ConnectionError` will
+        be raised (part of the `requests` library). If it fails during the web
+        request or JSON parsing, a `RemotePlatformDataException` is raised.
+        If it fails during the write to the filesystem, a `DatabaseUpdateException`
+        is raised.
+
+        Returns dictionary of all updates that occurred in the following format:
+            {
+                "new": {
+                    "<target id>": "<platform name>",
+                    ...
+                },
+                "updated": {
+                    "<target id>": {
+                        "old": "<previous platform name>",
+                        "new": "<new platform name>"
+                    }
+                }
+            }
+        """
         r = requests.get(url)
-        platform_data = r.json()
+        try:
+            platform_data = r.json()
```

Comment:
would be good if received json blob is schema validated before using json to avoid unexpected behaviour..

Reply:
Next PR?

Reply:
I do a bit of catching of schema errors on line 541. It prints to debug if the schema changes. I don't think we'll be doing anything stronger than that, except maybe raising a warning?